api/status_info.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+)
+
+// StatusInfo contains information about a user's status on a particular language track.
+type StatusInfo struct {
+	TrackID         string `json:"track_id"`
+	Recent          *Recent
+	FetchedProblems []string `json:"fetched"`
+	SkippedProblems []string `json:"skipped"`
+}
+
+// Recent contains information about the user's most recently submitted exercise on a particular language track.
+type Recent struct {
+	Problem     string `json:"problem"`
+	SubmittedAt string `json:"submitted_at"`
+}
+
+func (r *Recent) String() string {
+	submittedAt, err := time.Parse(time.RFC3339Nano, r.SubmittedAt)
+	if err != nil {
+		log.Fatal(err)
+	}
+	year, month, day := submittedAt.Date()
+
+	return fmt.Sprintf(" - %v (submitted on %v %v, %v)", r.Problem, month, day, year)
```

Comment:
This should use `%s` for the problem, and then also `%s` for the formatted time, once it's using a layout.